--- app/build.gradle.kts	2025-09-13 13:54:37
+++ app/build.gradle.kts.working	2025-09-13 13:52:51
@@ -4,17 +4,17 @@
 plugins {
     id("com.android.application")
     kotlin("android")
-    // // kotlin("kapt")
+    kotlin("kapt")
     id("de.mannodermaus.android-junit5")
     id("com.github.ben-manes.versions") version Dependencies.Versions.dependencyUpdates
 }
 
 android {
-    compileSdkVersion(33)
+    compileSdkVersion(30)
     defaultConfig {
         applicationId = "org.jellyfin.mobile"
         minSdkVersion(21)
-        targetSdkVersion(33)
+        targetSdkVersion(30)
         versionName = project.getVersionName()
         versionCode = getVersionCode(versionName!!)
         setProperty("archivesBaseName", "jellyfin-android-v$versionName")
@@ -85,13 +85,13 @@
     implementation(Dependencies.Kotlin.coroutinesAndroid)
 
     // Core
-    // // // // // // // // implementation(Dependencies.Core.koin)
-    // // // // // // // // implementation(Dependencies.Core.koinViewModel)
+    implementation(Dependencies.Core.koin)
+    implementation(Dependencies.Core.koinViewModel)
     implementation(Dependencies.Core.appCompat)
     implementation(Dependencies.Core.androidx)
     implementation(Dependencies.Core.activity)
     implementation(Dependencies.Core.fragment)
-    // // // // // // // // implementation(Dependencies.Core.koinFragment)
+    implementation(Dependencies.Core.koinFragment)
     implementation(Dependencies.Core.exoPlayer)
 
     // Lifecycle
@@ -110,10 +110,10 @@
 
     // Room
     implementation(Dependencies.Room.runtime)
-    // kapt(Dependencies.Room.compiler)
+    kapt(Dependencies.Room.compiler)
 
     // Network
-    // // // // // // // // implementation(Dependencies.Network.apiclient)
+    implementation(Dependencies.Network.apiclient)
     implementation(Dependencies.Network.okHttp)
     implementation(Dependencies.Network.coil)
     implementation(Dependencies.Network.exoPlayerHLS)
@@ -131,8 +131,8 @@
     // Health
     implementation(Dependencies.Health.timber)
     debugImplementation(Dependencies.Health.leakCanary)
-    // debugImplementation(Dependencies.Health.redScreenOfDeath)
-    // releaseImplementation(Dependencies.Health.redScreenOfDeathNoOp)
+    debugImplementation(Dependencies.Health.redScreenOfDeath)
+    releaseImplementation(Dependencies.Health.redScreenOfDeathNoOp)
 
     // Testing
     testImplementation(Dependencies.Health.junit)
